src/app/collective-rewards/shared/hooks/useGetABI.ts

```diff
@@ -123,15 +122,18 @@ export const useGetMetricsAbi = () => {
   const { cyclePayout, isLoading: cyclePayoutLoading, error: cyclePayoutError } = useGetCyclePayout()
 
   const rewardsPerStRif = useMemo(() => {
-    const sumTotalAllocation = Object.values(totalAllocation).reduce((acc, value) => acc + (value ?? 0n), 0n)
+    const sumTotalAllocation = Object.values(totalAllocation).reduce<bigint>(
+      (acc, value) => acc + (value ?? 0n),
+      0n,
+    )
 
     if (!sumTotalAllocation) {
       return 0n
     }
 
     const topFiveBuilders = builders
-      .reduce<Array<{ allocation: bigint; current: bigint }>>((acc, builder) => {
-        const allocation = totalAllocation[builder.gauge]
+      .reduce<Array<{ allocation: bigint; current: bigint }>>((acc, builder, i) => {
+        const allocation = totalAllocation[i]
```

Comment:
we already tried to keep track of the allocations with the index and we realised it was better to use the gauge address explicitly

Reply:
It doesn't apply in this specific context.